### ISSUE TITLE
feat: Switch `&str` params

### ISSUE THREAD

**Issue by HarryET** [WalletConnect/a2#67]
Stop using `&str` with lifetime for the params and instead use `impl Into<String>`

**Comment by MohammadAbuzar945** [comment]
Using i`mpl Into<String>` allows you to accept any type that can be converted into a `String`. This can be more flexible than using `&str` with a lifetime, because it allows the caller to pass in a variety of types, including `owned Strings, &String, &str,` and even types like `String, Vec,` and `Box` that implement the `Into `trait.

**Comment by HarryET** [comment]
Yes, that is why we have this issue open to make that change, is it something you want to do @MohammadAbuzar945?

**Comment by rex-remind101** [comment]
@HarryET This would imply removing the borrow parameter `'a` and moving to using `String` in `Payload` correct? https://github.com/WalletConnect/a2/blob/0b29470b6d559a9b3ae8d7136151b0a8e32f60b3/src/request/payload.rs#L10

**Comment by HarryET** [comment]
> @HarryET This would imply removing the borrow parameter `'a` and moving to using `String` in `Payload` correct?
> 
> https://github.com/WalletConnect/a2/blob/0b29470b6d559a9b3ae8d7136151b0a8e32f60b3/src/request/payload.rs#L10

Correct, it should use `String` internally and all params should be `S: Into<String>` as a generic

**Comment by FrenchGithubUser** [comment]
Hey, if no one is on it, I will do this!

**Comment by HarryET** [comment]
Hey! I'm no-longer at reown; \cc @chris13524 might be able to take a look?
